### DISCUSSION TITLE
Hiding an edge when either source or target node is hidden

### DISCUSSION THREAD

**Discussion by ronrosenman** [xyflow/xyflow#4204] [Q&A]
Hello! My first time asking a question here. Great library!

I am trying to hide an edge when either its source or target node is hidden, I thought https://github.com/xyflow/xyflow/discussions/4130 might have an answer but the construction there is slightly different than mine

I have a boolean, xNodesHidden, that gets set in the ui via a checkbox and then inside the flow object I have 

```
const NestedFlow = () => {. 
  ...  
  useEffect(() => {
    setNodes((nds) =>
      nds.map(node) =>
        if (node.type === 'x') {
          node.hidden = xNodesHidden;
        }
        return node;  
     })
  );  
}, [xNodesHidden, setNodes]);  
```

so when I check the box, all the nodes of type x are hidden, but I would like this action to also hide all of the edges leading in or out of those nodes.

Is there a good existing example to follow?

Thanks

**Comment by bcakmakoglu** [comment]
You could simply go through the edges and hide them  as well  :)

```ts
  useEffect(() => {
    setNodes((nds) =>
      nds.map(node) =>
        if (node.type === 'x') {
          node.hidden = xNodesHidden;
        }
        return node;  
     })

	setEdges(edges =>
		edges.map(edge => {
			// getNode is a helper function available from `useReactFlow`
			const sourceNode = getNode(edge.source)
			const targetNode = getNode(edge.target)

			if (sourceNode.type === 'x' || targetNode.type === 'x') {
				return { ...edge, hidden: true }
			}
	
			return edge;
		}
  );  
}, [xNodesHidden, setNodes, setEdges]);
